pkgs/by-name/wi/widevine-cdm/update-x86_64.py

```diff
@@ -0,0 +1,148 @@
+        return entry["fileUrl"]
+
+    m = entry.get("mirrorUrls")
+    if isinstance(m, list) and m:
+        return m[0]
+
+    return None
+
+def extract_ver_from_url(url: str) -> Optional[str]:
+    m = re.search(r'[_-](\d+\.\d+\.\d+\.\d+)[_-]', url)
+    if m:
+        return m.group(1)
+    m = re.search(r'(\d+\.\d+\.\d+\.\d+)', url)
+    return m.group(1) if m else None
+
+def build_entry(pkey: str, pentry: dict) -> dict:
+    url = normalize_fileurl(pentry)
+    hv = pentry.get("hashValue")
+
+    if not url or not hv:
+        raise SystemExit(f"ERR: platform {pkey} is missing a file URL or hash !")
+
+    sri = hex_to_sri(hv)
+    version = extract_ver_from_url(url)
+    return {"platform_key": pkey, "url": url, "hashValue": hv, "sri": sri, "version": version}
+
+
+
+def main():
+    ap = argparse.ArgumentParser()
+    ap.add_argument("--json-url", default="https://raw.githubusercontent.com/mozilla-firefox/firefox/refs/heads/main/toolkit/content/gmp-sources/widevinecdm.json")
+    args = ap.parse_args()
+
+    print(f"# fetching {args.json_url} !", file=sys.stderr)
+    data = fetch_json(args.json_url)
+    vendor = find_widevine_vendor(data)
+    platforms = vendor.get("platforms") or {} # should never be null but moz could forseeably delete it
+    if not platforms:
+        raise SystemExit("ERR: no widevine platforms !")
+
+    results = {}
+    for arch, patterns in ARCH_PATTERNS.items():
+        found = judge_platforms(platforms, patterns)
+        if not found:
+            print(f"# WARN: no platform for {arch}. skipping !", file=sys.stderr)
+            continue
+        pkey, pentry = found
+        print(f"# found {arch} --> {pkey}", file=sys.stderr)
+        entry = build_entry(pkey, pentry)
+        results[arch] = entry
+
+    if not results:
+        raise SystemExit("ERR: no linux platform entries !")
+
+    linux_x64 = results["linux_x86_64"]
+
+    try:
+        with open("x86_64-linux.nix", "w") as f:
+            f.write(f"""{{
+  lib,
+  stdenv,
+  fetchurl,
+  go-crx3,
+}}:
+
+stdenv.mkDerivation (finalAttrs: {{
+  # Generated by update.py (don't edit this file manually)
+  pname = "widevine-cdm";
+  version = "{linux_x64["version"]}";
+
+  src = fetchurl {{
+    url = "{linux_x64["url"]}";
+    hash = "{linux_x64["sri"]}";
+  }};
+
+  nativeBuildInputs = [ go-crx3 ];
+
+  unpackPhase = ''
+    unpackDir="src"
+    cp "$src" "$unpackDir".crx # go-crx3 doesn't like .crx3 extensions
+    crx3 unpack "$unpackDir".crx
+
+    cd "$unpackDir"
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -vD manifest.json $out/share/google/chrome/WidevineCdm/manifest.json
+    install -vD LICENSE $out/share/google/chrome/WidevineCdm/LICENSE.txt
+    install -vD _platform_specific/linux_x64/libwidevinecdm.so $out/share/google/chrome/WidevineCdm/_platform_specific/linux_x64/libwidevinecdm.so
+
+    runHook postInstall
+  '';
+
+  meta = import ./meta.nix lib;
+  passthru.updateScript = ./update-x86_64.py;
+}})""")
```

Comment:
Update scripts should not generate nix files like that as it is very likely that with the next update we need to retrofit changes to the update script which is just a churn. If replacing the hashes and download URLs is not possible with simple regex's we can always write a simple nix or json file to import it. But just replacing ``version = ".*"`` and likewise url and hash should not be that difficult.